src/main/java/org/socialhistoryservices/delivery/reservation/controller/ReservationController.java

```diff
@@ -216,16 +225,28 @@ private String processVisitorReservationCreation(HttpServletRequest req, Reserva
                     // Automatically print the reservation.
                     autoPrint(newRes);
                     model.addAttribute("reservation", newRes);
+
+
+                    // set all available holdings in the same container as blocked
+                    ArrayList<String> listOfPids = getAllAvailableHoldingsInReservation(newRes);
+                    List<Holding> holdingList = getHoldingsWithSpecificStatus(listOfPids, Holding.Status.AVAILABLE);
+                    if (holdingList != null) {
+                        for (Holding hld : holdingList) {
+                            hld.setStatus(Holding.Status.RESERVED);
```

Comment:
Don't do this on the controller level. This doesn't happen for the regular holdings of the reservation either.